osu.Game/Overlays/MusicController.cs

```diff
@@ -389,11 +392,17 @@ public void ResetTrackAdjustments()
             CurrentTrack.RemoveAllAdjustments(AdjustableProperty.Frequency);
             CurrentTrack.RemoveAllAdjustments(AdjustableProperty.Tempo);
             CurrentTrack.RemoveAllAdjustments(AdjustableProperty.Volume);
+            audioManager.TrackMixer.Effects.Clear();
+            audioManager.SampleMixer.Effects.Clear();
 
             if (allowTrackAdjustments)
             {
                 foreach (var mod in mods.Value.OfType<IApplicableToTrack>())
                     mod.ApplyToTrack(CurrentTrack);
+                foreach (var mod in mods.Value.OfType<IApplicableToTrackMixer>())
+                    mod.ApplyToTrackMixer(audioManager.TrackMixer);
+                foreach (var mod in mods.Value.OfType<IApplicableToSampleMixer>())
+                    mod.ApplyToSampleMixer(audioManager.SampleMixer);
```

Comment:
> - **Should IApplicableToTrack provide the mixer instead of a separate interface?**
@frenzibyte originally proposed to include the mixer in IApplicableToTrack.ApplyToTrack, but there are many cases where a mixer isn't available (e.g. when using this interface to calculate new track rate in diff calc), so I decided to split this into a new interface.

I still don't really see the reason behind these existing, you can add the mixer but as an optional parameter for such cases:
```csharp
public void ApplyToTrack(ITrack track, Mixer mixer = null);
```

And avoid all the extra additions of `IApplicableToTrackMixer`/`IApplicableToSampleMixer`/`IApplicableToMixers`.

Reply:
Why would the mixer be nullable in the above proposal? What would a mod implementation do if it is supplied a null mixer while it wants to apply effects, just give up?

I would even go a step further and propose that maybe `ITrack` should itself provide a reference to the mixer, rendering all of this unnecessary. But I don't feel confident on that proposal, so I'd advise not to take it too seriously.

Reply:
> Why would the mixer be nullable in the above proposal? What would a mod implementation do if it is supplied a null mixer while it wants to apply effects, just give up?

It's nullable because it's irrelevant in cases like diffcalc. And, correct, I don't see that being a problem, we could not make it an optional parameter if your concern is about it being forgotten in future usages?

Reply:
Yeah I dunno. I think diffcalc doing stuff on tracks in general is kinda dodgy already, and it's currently worked around by passing a virtual track. I'd rather follow that lead and pass a dummy mixer or something, but the perfect scenario would be to decouple diffcalc from audio stuff.

Reply:
Having a `VirtualAudioMixer` sounds workable, sure.

Reply:
Created a framework PR for that: https://github.com/ppy/osu-framework/pull/5013